src/Version.cpp

```diff
@@ -9,22 +9,58 @@
 
 namespace Version {
 
-unsigned int getRaw(const QString &version) {
-	int major, minor, patch;
+QString getRelease() {
+	return QLatin1String(MUMBLE_RELEASE);
+}
+
+Version::mumble_raw_version_t getRaw(const QString &version) {
+	Version::component_t major, minor, patch;
 
-	if (get(&major, &minor, &patch, version))
+	if (get(&major, &minor, &patch, version)) {
 		return toRaw(major, minor, patch);
+	}
 
-	return 0;
+	return Version::UNKNOWN;
 }
 
-QString toString(unsigned int v) {
-	int major, minor, patch;
+Version::mumble_raw_version_t fromConfig(const QVariant &config) {
+	Version::mumble_raw_version_t version;
+
+	QRegExp digitTest("\\d+");
```

Comment:
What if it's given as a hexadecimal number?
Why don't you simply use `toULongLong` and see if that is != 0? That seems more straight forward than using RegEx for this 👀

Reply:
I think ``\d`` will match only decimals. Was hex also supported? The reason for this instead of ``toULongLong`` is that uLongLong makes a premature assumption of the size of the version number. I would use QBigInt if such a thing existed... But I think this way it is clear that the ``toULongLong`` only applies to the current version format... But, I don't know... The other way works, too of course. The other way just feels slightly ugly to me...

Reply:
> I think \d will match only decimals. 

Yes indeed

> Was hex also supported?

I would have assumed so as that's how the bitmasked version is represented everywhere else...

> The reason for this instead of toULongLong is that uLongLong makes a premature assumption of the size of the version number.

How so? A `long long` should always be big enough to hold the new version format integer and thus also the old version format integer. It night be bigger tham necessary but that doesn't seem to be an issue?

Reply:
> I would have assumed so as that's how the bitmasked version is represented everywhere else...

Looking at the old code I am pretty sure that it could not automatically convert hex values, since the integer was loaded directly into the version variable without base 16 conversion... So to represent the version 1.2.3 you would have to either write ``1.2.3`` or ``66051`` into your settings file...

Reply:
> How so? A `long long` should always be big enough to hold the new version format integer and thus also the old version format integer. It might be bigger tham necessary but that doesn't seem to be an issue?

It is indeed not a problem right now. We could easily use toULongLong. It is just a little ugly (and the toUInt was also a little ugly before). I tried to deliberately remove all instances where an assumption of the size of ``mumble_raw_version_t`` is made outside of the methods which create it. It just feels wrong to call toULongLong to check if the user string is a valid version integer... It screams "This will regress in the future" to me...

But I can use toULongLong no problem, if you think otherwise

Reply:
@Krzmbrzl Looking at #989 I am very sure this never allowed for hex strings... And because an bitmasked integer in decimal is so very unintutive, I would rather have dropped the feature entirely in favor of simple strings...

Reply:
> Looking at the old code I am pretty sure that it could not automatically convert hex values, since the integer was loaded directly into the version variable without base 16 conversion

I'm reasonably sure, Qt performs this conversion automatically when calling `toUInt` on a `QVariant` that holds a String that could be read as a hexadecimal number.

> I tried to deliberately remove all instances where an assumption of the size of mumble_raw_version_t is made outside of the methods which create it.

Well, the current code still assumes that every numeric input fits in a `std::uint64_t`, so I don't see a big difference in trying that from the beginning (other than losing any potential implicit conversions (e.g. from hex) that Qt performs when accessing the variant.

> Looking at #989 I am very sure this never allowed for hex strings

Ah yes indeed.

> And because an bitmasked integer in decimal is so very unintutive, I would rather have dropped the feature entirely in favor of simple strings

Again: This PR is supposed to get backported, so dropping functionality is not an option

Reply:
I changed it, but for future reference a hypothetical ``toBigInt`` would have been the best way to do it ;)